gui/js/lastfiles.js

```diff
@@ -5,15 +5,22 @@
 
 const fs = require('fs')
 const path = require('path')
+const async = require('async')
 
 let lastFilesPath = ''
 let lastFiles = Promise.resolve([])
+let persistQueue = null
 
 const log = require('../../core/app').logger({
   component: 'GUI'
 })
 
+const writeJSON = (data, callback) => {
+  fs.writeFile(lastFilesPath, data, callback)
+}
+
 const init = desktop => {
+  persistQueue = Promise.promisifyAll(async.queue(writeJSON))
```

Comment:
Why are we using `promisifyAll` here?

Reply:
So that the `queue.push` call returns a promise (`queue.pushAsync` to be exact).
I'd prefer not to deal with callbacks.

Reply:
I might be misreading it, but to me it sounds like `pushAsync` returns a promise anyway https://caolan.github.io/async/v3/docs.html#QueueObject ?

Reply:
We're using `async` v2.6.2 which does not have those promisified methods.
We should update the dependency at some point.

Reply:
aha, gotcha!